config/src/main/java/com/typesafe/config/impl/ConfigImpl.java

```diff
@@ -364,15 +364,37 @@ public static Config defaultReference(final ClassLoader loader) {
         return computeCachedConfig(loader, "defaultReference", new Callable<Config>() {
             @Override
             public Config call() {
-                Config unresolvedResources = Parseable
-                        .newResources("reference.conf",
-                                ConfigParseOptions.defaults().setClassLoader(loader))
-                        .parse().toConfig();
+                Config unresolvedResources = unresolvedReference(loader);
                 return systemPropertiesAsConfig().withFallback(unresolvedResources).resolve();
             }
         });
     }
 
+    private static Config unresolvedReference(final ClassLoader loader) {
+        return computeCachedConfig(loader, "unresolvedReference", new Callable<Config>() {
+            @Override
+            public Config call() {
+                return Parseable.newResources("reference.conf",
+                        ConfigParseOptions.defaults().setClassLoader(loader))
+                    .parse().toConfig();
+            }
+        });
+    }
+
+    /**
+     * This returns the unresolved reference configuration, but before doing so,
+     * it verifies that the reference configuration resolves, to ensure that it
+     * is self contained and doesn't depend on any higher level configuration
+     * files.
+     */
+    public static Config verifiedUnresolvedReference(final ClassLoader loader) {
```

Comment:
@havocp So we have this as public API, would you like it in `ConfigFactory`, with an overload similar to `defaultReference`? Is the name ok?

Reply:
I was thinking of the name `defaultReference` as "literally the reference used by default in ConfigFactory.load", paralleled by `defaultApplication` and `defaultOverrides` ; we do need to break that meaning to do this PR, which is OK, but I wonder if I've sort of encoded in the docs in spots.

A possible name here could be `defaultReferenceUnresolved` which would kind of keep the parallelism showing this is part of the default stack, but with the tweak that there's an unresolved version.

I think verified is a bit of an implementation detail and not the main point of the method.

I do think I'd put it in ConfigFactory; it's fairly common for people to slightly tweak `ConfigFactory.load` in some way, so that's why its individual "component parts" are made available, trying to make it easy to rebuild `load` with tweaks.

Reply:
Ok, I've renamed as requested, and copied it to `ConfigFactory`.